### PR TITLE
Replace deprecated func usage from the `k8s.io/utils/pointer` pkg

### PR DESCRIPTION
**Pull request by ialidzhikov**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Some of the funcs in the `k8s.io/utils/pointer` pkg were deprecated with https://github.com/kubernetes/utils/pull/249. This PR adapts the kubernetes/kubernetes codebase to no longer use the deprecated funcs and instead to use their equivalent.

#### Which issue(s) this PR fixes:

N/A

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: pkg/scheduler/framework/plugins/nodevolumelimits/non_csi_test.go [hunk 1]
```diff
@@ -792,14 +792,14 @@ func getFakePVCLister(filterName string) fakeframework.PersistentVolumeClaimList
 			ObjectMeta: metav1.ObjectMeta{Name: "unboundPVCWithDefaultSCPod"},
 			Spec: v1.PersistentVolumeClaimSpec{
 				VolumeName:       "",
-				StorageClassName: utilpointer.StringPtr("standard-sc"),
+				StorageClassName: utilpointer.String("standard-sc"),
```
Comment (alculquicondor) [inline pkg/scheduler/framework/plugins/nodevolumelimits/non_csi_test.go:795]: take the chance to remove the alias from the package. `pointer` is enough.

File: pkg/util/tolerations/tolerations_test.go [hunk 2]
```diff
@@ -99,20 +99,20 @@ var (
 			Key:               "foo",
 			Operator:          api.TolerationOpExists,
 			Effect:            api.TaintEffectNoExecute,
-			TolerationSeconds: utilpointer.Int64Ptr(10),
+			TolerationSeconds: utilpointer.Int64(10),
```
Comment (alculquicondor) [inline pkg/util/tolerations/tolerations_test.go:102]:
```suggestion
			TolerationSeconds: pointer.Int64(10),
```

File: pkg/volume/csi/nodeinfomanager/nodeinfomanager_test.go [hunk 3]
```diff
@@ -459,7 +459,7 @@ func TestInstallCSIDriver(t *testing.T) {
 							NodeID:       "com.example.csi/csi-node1",
 							TopologyKeys: nil,
 							Allocatable: &storage.VolumeNodeResources{
-								Count: utilpointer.Int32Ptr(10),
+								Count: utilpointer.Int32(10),
```
Comment (alculquicondor) [inline pkg/volume/csi/nodeinfomanager/nodeinfomanager_test.go:462]:
```suggestion
								Count: pointer.Int32(10),
```

File: plugin/pkg/admission/network/defaultingressclass/admission_test.go [hunk 4]
```diff
@@ -134,30 +134,30 @@ func TestAdmission(t *testing.T) {
 			classes:         []*networkingv1.IngressClass{defaultClass1, classWithFalseDefault, classWithNoDefault, classWithEmptyDefault},
 			classField:      nil,
 			classAnnotation: nil,
-			expectedClass:   utilpointer.StringPtr(defaultClass1.Name),
+			expectedClass:   utilpointer.String(defaultClass1.Name),
```
Comment (alculquicondor) [inline plugin/pkg/admission/network/defaultingressclass/admission_test.go:137]: pointer

**Review by alculquicondor — COMMENTED** [review]
/approve
for scheduling... however, we added a few new files since you opened this PR. Please rebase and fix those too.

**Comment by ialidzhikov** [comment]
@alculquicondor , thank you for your review.
I think it will be better if I split this PR to n smaller PRs for each pkg in https://github.com/kubernetes/kubernetes/pull/111585#issuecomment-1204349553. Otherwise I don't see how I could get approved + lgtm on this PR from all pkg owners (except some approval from root level).

/close

**Pull request closed without merging**